CMakeLists.txt

```diff
@@ -272,10 +272,14 @@ TARGET_COMPILE_DEFINITIONS(wxWidgets INTERFACE ${wxWidgets_DEFINITIONS}
 # --------- curl dependency ---------
 FIND_PACKAGE(CURL QUIET COMPONENTS libcurl CONFIG)
 IF(CURL_FOUND)
-    MESSAGE(STATUS "Found CURL: ${CURL_LIBRARIES} (found version \"${CURL_VERSION}\")")
+    MESSAGE(STATUS "Found CURL version: \"${CURL_VERSION}\")")
+    MESSAGE(STATUS "CURL Libraries: ${CURL_LIBRARIES}")
+    MESSAGE(STATUS "CURL Include Directories: ${CURL_INCLUDE_DIRS}")
     SET(CURL_VERSION_STRING ${CURL_VERSION})
 ELSE()
     FIND_PACKAGE(CURL REQUIRED COMPONENTS libcurl MODULE)
+    MESSAGE(STATUS "CURL Module Libraries: ${CURL_LIBRARIES}")
```

Comment:
For a CURL installation on Windows, the CURL solution files can be generated using the CMake GUI. The libs can be built and the install solution needs to be run separately to create the curl/install files. Now the path to curl/install/CMake needs to be added to the system path list for curl to be found. To ensure that this has been set up correctly, the CURL Include Directories will be displayed in the configuration. If not, the CURL Include Directories will show- Not found
![screenshot 2018-02-27 13 29 29](https://user-images.githubusercontent.com/6927151/36707358-d9ea0512-1b65-11e8-9a2a-c5eba8cab507.png)

These messages in the CMake script are added to ensure correct installation of CURL


This now shows the correct installation of CURL for mmex
![screenshot 2018-02-27 13 34 09](https://user-images.githubusercontent.com/6927151/36707555-b7fe30d0-1b66-11e8-8094-07f93c41edd7.png)

Reply:
> Now the path to curl/install/CMake needs to be added to the system path list for curl to be found.

This is not true.

You are trying to set cmake [internal cache entry `CURL_DIR`](https://cmake.org/cmake/help/v3.11/manual/cmake-packages.7.html#config-file-packages) to wrong value. This cache entry is used to store path to curl library discovered in previous run. This must point to installed library sub-dir. When you set this to your build-tree it will not work as you can see.

Please don't use CURL_DIR anymore. There is enough logic in cmake to detect all required paths for libcurl installed (on different platform, architecture, configuration etc.) using only standard [`CMAKE_PREFIX_PATH`](https://cmake.org/cmake/help/v3.11/variable/CMAKE_PREFIX_PATH.html)

1. Store curl sources in `c:\curl-src`
2. Configure curllib in `c:\curl-src\build` with `CMAKE_INSTALL_PREFIX=c:\curl-lib`
3. Build and install curllib from `c:\curl-src\build` - this will install all required binaries, includes and config files to `c:\curl-lib`
4. Configure MMEX project with `CMAKE_PREFIX_PATH=c:\curl-lib`

Please reset your cmake cache entries (with wrong CURL_DIR values) before trying above (or simply remove your build subdir).

BTW you should upgrade your cmake 3.9.x to 3.10 or even 3.11.

Reply:
When starting from new build location or delete cache, when the path is not set, Curl is not found.
This also upsets the mmex install location and puts to default to C:\Program Files\mmex
This was my initial problem in trying to get this working.

Reply:
This version only supplies me with CURL_DIR. Perhaps newer versions will supply other parameters that can be modified. As you can see from above pictures when this is set correctly manually if not found, or automatically if the path is set up first, then the correct project files can be created.

Reply:
As for using C:\curl-src I do not like putting a lot of folders in the root folder

Reply:
All cmake 3.x.x versions support this mechanism.

`<MODULE>_DIR` cache entry cannot be used this way. It should store last found and *working* location.

`CURL_DIR=C:\Git\curl\install\CMake` should be replaced with `CMAKE_PREFIX_PATH=C:\Git\curl\install` in your case. This will use proper config file from

C:\Git\curl\install\
C:\Git\curl\install\cmake\
C:\Git\curl\install\CMake\   <-
C:\Git\curl\install\curl*\
C:\Git\curl\install\curl*\cmake\
C:\Git\curl\install\curl*\CMake\
C:\Git\curl\install\curl*\lib\x86\cmake\curl*\
C:\Git\curl\install\curl*\lib\x64\cmake\curl*\
C:\Git\curl\install\curl*\lib\cmake\curl*\
C:\Git\curl\install\curl*\share\cmake\curl*\
C:\Git\curl\install\curl*\lib\x86\curl*\
C:\Git\curl\install\curl*\lib\x64\curl*\
C:\Git\curl\install\curl*\lib\curl*\
C:\Git\curl\install\curl*\share\curl*\
C:\Git\curl\install\curl*\lib\x86\curl*\cmake\
C:\Git\curl\install\curl*\lib\x64\curl*\cmake\
C:\Git\curl\install\curl*\lib\curl*\cmake\
C:\Git\curl\install\curl*\share\curl*\cmake\
C:\Git\curl\install\curl*\lib\x86\curl*\CMake\
C:\Git\curl\install\curl*\lib\x64\curl*\CMake\
C:\Git\curl\install\curl*\lib\curl*\CMake\
C:\Git\curl\install\curl*\share\curl*\CMake\

You can use any dirs in place of `c:\curl-src`, `c:\curl-src\build` and `c:\curl-lib` of course.

Please follow the 4 steps above (and examples in BUILD.md and appveyor.yml) without manually modifying cmake scripts and use `CMAKE_INSTALL_PREFIX` and `CMAKE_PREFIX_PATH` only to set curl paths and report any issues.

Reply:
I take it that I need to use dos first?